plugins/lookup/merge_variables.py

```diff
@@ -0,0 +1,194 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, Thales Netherlands
+# Copyright (c) 2021, Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+    author:
+      - Roy Lenferink (@rlenferink)
+      - Mark Ettema (@m-a-r-k-e)
+    name: merge_variables
+    short_description: merge variables with a certain suffix
+    description:
+        - This lookup returns the merged result of all variables in scope that matches the given suffix optionally
+          starting with an initial value.
+    version_added: 6.1.0
+    options:
+      _terms:
+        description:
+          - The suffix of the variable name (all available variables will be checked with endswith(suffix))
+        required: true
+      pattern_type:
+        description:
+          - Change the way of searching for the specified pattern.
+          - Possible values are C(prefix), C(suffix) and C(regex). By default C(suffix) is used.
+        type: str
+        default: 'suffix'
+        env:
+          - name: ANSIBLE_MERGE_VARIABLES_PATTERN_TYPE
+      initial_value:
+        description:
+          - An initial value to start with
+        required: false
+      override:
+        description:
+          - Return an error, print a warning or ignore it when a key will be overwritten.
+          - Possible values are C(error), C(warn) and C(ignore). By default C(error) is used.
```

Comment:
Since that information is already provided by the below metadata, why not add some more information instead?
```suggestion
          - The default behavior C(error) makes the plugin fail when a key would be overwritten.
          - When C(warn) and C(ignore) are used, note that it is important to know that the variables
            are sorted by name before being merged. Keys for later variables in this order will overwrite
            keys of the same name for variables earlier in this order. To avoid potential confusion,
            better use I(override=error) whenever possible.
```

Reply:
Adopted your suggestion